### ISSUE TITLE
Error message for get_artist_audio_features()

### ISSUE THREAD

**Issue by temporaldrift** [charlie86/spotifyr#59]
When I try to run get_artist_audio_features(), I get the error message:
```
Error in handle_url(handle, url, ...) : 
  could not find function "str_glue"
```
I already have tidyverse and stringr loaded; is there anything else I should be doing to make sure the str_glue function is available? Thanks!

**Comment by antaldaniel** [comment]
There are several problems with this function, and I did correct at least one.  If there is an agreement, I would rewrite the whole tracks.R. 

1. The functions inconsistently use tracks inputs.  get_track_audio_analysis() uses tracks as a string input, and checks there validity, which is a good practice.
2. Some other functions use a tibble or data.frame input from get_tracks() which reduces usability - I want to get features of my own list - and is entirely superfluous given that only the character string of track URIs is needed. My solution (pull request) currently checks if the input is a character vector or a tibble / data.frame, but I think it would be easier to have an internal function that retrieves the character vector from the get_tracks tibble, when needed, otherwise allows string input that is verified. 
3. For this reason I made a copy of the is_uri function and string cleaning a separate check_uri internal function which is not exported.
4. Generally I do not find it good practice to not explicitly import all dependency functions of a function, like str_glue above.  If you agree with this, I would rewrite tracks.R to make all dependencies explicit, to avoid further debugging problems.
5. I started improving the documentation because internal links are missing to referenced functions.

I made a pull request only  on get_track_audio_features(), a function that also failed me. With audio_features <- get_track_audio_features('0bG6Z71CiduR9VBUaCfwzp'), and of course my api settings it works for me.
